sympy/combinatorics/perm_groups.py

```diff
@@ -4861,6 +4861,44 @@ def polycyclic_group(self):
 
         return PolycyclicGroup(pc_sequence, pc_series, relative_order, collector=None)
 
+    def exponent(self):
+        """
+            Returns the exponent of a group
+
+            The exponent e of a group G is the lcm of the orders of its elements,
+            that is, e is the smallest integer such that g^e = 1 for all g belong to G.
+            or the exponent of a group G is just the product of exponents of its sylow_subgroups.
+
+            Examples
+            ========
+
+            >>> from sympy.combinatorics import Permutation
+            >>> from sympy.combinatorics.perm_groups import PermutationGroup
+            >>> a = Permutation(1, 2)(2, 3)
+            >>> b = Permutation(1, 2, 3, 4)(2, 3, 4)(5, 6, 7)(8, 9)
+            >>> A = PermutationGroup([a, b])
+            >>> A.exponent()
+            12
+            >>> c = Permutation(1, 2, 3, 4, 5, 6)(2, 3, 4)(5, 6, 7)(8, 9)(10, 11, 12)
+            >>> A = PermutationGroup([a, b, c])
+            >>> A.exponent()
+            420
+
+        """
+
+        exp = 1
+        sylow_groups = {}
+        P = factorint(self.order()).keys()
+        for p in P:
+            sylow_p_subgroup = self.sylow_subgroup(p)
+            if sylow_p_subgroup not in sylow_groups:
+                elements = list(sylow_p_subgroup.elements)
+                exp_p = elements[0].order()
+                for i in range(1, len(elements)):
```

Comment:
This looks inefficient. `len(elements)` may be very big.

Reply:
Okay, got it 
I looked at the gap implementation that method will require the computation of many other methods too such `Leedham-Green series` and `special pcgs` and many others too.

should I go ahead and implement those in new pr's and after that jump to this pr again?

Reply:
It seems that something like that will be needed. If there are no algorithms in the Handbook, then those method may be hard to implement.

Reply:
Yes you are right It can be hard probably I can take help of gap implementation but implementing these may take too much time so first I will try to open a pr on which I have already worked and they are on my system.